From 7d4f30f63b8aff10ee92d7e23347e763fcaf7806 Mon Sep 17 00:00:00 2001
From: Double Lo <double.lo@cypress.com>
Date: Sun, 18 Aug 2019 22:31:19 -0500
Subject: [PATCH 117/154] brcmfmac: 43012 Update MES Watermark

Set MES watermark size to 0x50 for 43012. It fixes SDIO bus hang issue
when running at high throughput.

Signed-off-by: Double Lo <double.lo@cypress.com>
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
index 359502f..1b4014f 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
@@ -55,6 +55,9 @@
 #define CY_4373_F2_WATERMARK    0x40
 #define CY_4373_F1_MESBUSYCTRL  (CY_4373_F2_WATERMARK | SBSDIO_MESBUSYCTRL_ENAB)
 #define CY_43012_F2_WATERMARK    0x60
+#define CY_43012_MES_WATERMARK  0x50
+#define CY_43012_MESBUSYCTRL    (CY_43012_MES_WATERMARK | \
+				 SBSDIO_MESBUSYCTRL_ENAB)
 #define CY_43455_F2_WATERMARK	0x60
 #define CY_43455_MES_WATERMARK	0x50
 #define CY_43455_MESBUSYCTRL	(CY_43455_MES_WATERMARK | \
@@ -4373,6 +4376,8 @@ static void brcmf_sdio_firmware_callback(struct device *dev, int err,
 			/* Register for ULP events */
 			brcmf_fweh_register(bus_if->drvr, BRCMF_E_ULP,
 					    brcmf_ulp_event_notify);
+			brcmf_sdiod_regwb(sdiodev, SBSDIO_FUNC1_MESBUSYCTRL,
+					  CY_43012_MESBUSYCTRL, &err);
 			break;
 		case SDIO_DEVICE_ID_BROADCOM_43455:
 			brcmf_dbg(INFO, "set F2 watermark to 0x%x*4 bytes for 43455\n",
-- 
1.9.1

